jose/backends/rsa_backend.py

```diff
@@ -83,6 +86,59 @@ def pem_to_spki(pem, fmt='PKCS8'):
     return key.to_pem(fmt)
 
 
+def _legacy_private_key_pkcs8_to_pkcs1(der_key):
+    """Legacy RSA private key PKCS8-to-PKCS1 conversion.
+
+    .. warning::
+
+        This is incorrect parsing and only works because the legacy PKCS1-to-PKCS8
+        encoding was also incorrect.
+    """
+    return der_key[len(LEGACY_INVALID_PKCS8_RSA_HEADER):]
```

Comment:
Please add a check that the prefix of this invalid key is, in fact, equal to `LEGACY_INVALID_PKCS8_RSA_HEADER`, else reject.

Reply:
Check added looking for the legacy header followed immediately by an ASN1 sequence identifier. That will protect against the case where an otherwise-invalid key is processed here, since for 2048-bit keys the legacy prefix is actually correct, but incomplete.

Reply:
LGTM